eacl/validator.go

```diff
@@ -152,8 +152,14 @@ func targetMatches(unit *ValidationUnit, record *Record) bool {
 		// check public key match
 		if pubs := target.BinaryKeys(); len(pubs) != 0 {
 			for _, key := range pubs {
-				if bytes.Equal(key, unit.key) {
-					return true
+				if len(key) == 33 { // pub key.
+					if bytes.Equal(key, unit.key) {
+						return true
+					}
+				} else { // account (script hash).
+					if bytes.Equal(key, unit.account) {
```

Comment:
i'm worried about the case when `unit.account` is unset and `key` is empty (afaik this is not prohibited now). Suggest
```suggestion
					if len(unit.account) > 0 && bytes.Equal(key, unit.account) {
```